app/uk/gov/hmrc/bindingtariffclassification/repository/SearchMapper.scala

```diff
@@ -48,7 +48,7 @@ class SearchMapper @Inject()(appConfig: AppConfig) {
     val params = Seq[Option[(String, JsValue)]](
       filter.reference.map("reference" -> inArray[String](_)),
       filter.applicationType.map(filteringByApplicationType),
-      filter.queueId.map("queueId" -> mappingNoneOrSome(_)),
+      filter.queueId.map("queueId" -> inArrayOrNone[String](_)),
```

Comment:
To handle the case where we pass in a set containing both "none" and "some" we should not include the queueId filter.

```suggestion
      filter.queueId
        .filterNot(ids => ids.contains("some") && ids.contains("none"))
        .map("queueId" -> inArrayOrNone[String](_)),
```